src/model.rs

```diff
@@ -159,6 +160,21 @@ impl Collections {
             }
         }
 
+        fn dedup_collection<T: Clone + Eq + Hash>(source: &mut Collection<T>) -> Collection<T> {
+            let calculate_hash = |t: &T| -> u64 {
```

Comment:
`vec.dedup()` to avoid clone , :thinking: 
https://doc.rust-lang.org/std/vec/struct.Vec .html#method.dedup